### PR TITLE
st7789: add support for framerate, vsync pause, and scanline timing

### PR DESCRIPTION
**Pull request by jmstriegel**
- FrameRate option sets device framerates from 39-111Hz
- VSyncLines option adjusts device vsync pause using st7789 "porch control" feature
- Added Rx(cmd, bytes[]) to support retrieving scanline timing data over SPI
- Added Sync functions to support synchronization of animation to vertical scanline timing
- Minor adjustments to Configure to clear screen memory before display is visible

### CODE REVIEW

File: examples/st7789/main.go [hunk 1]
```diff
@@ -9,12 +9,37 @@ import (
...
+	//	SCK:       machine.TFT_SCK,
+	//	SDO:       machine.TFT_SDO,
+	//	SDI:       machine.TFT_SDO,
+	//	Mode:      0,
+	// })
+	// display := st7789.New(machine.SPI1,
+	//	machine.TFT_RESET,
+	//	machine.TFT_DC,
+	//	machine.TFT_CS,
+	//	machine.TFT_LITE)
```
Comment (jmstriegel) [inline examples/st7789/main.go:25]: Is there a best practice for handling hardware/machine specific recommendations in the device examples?

Comment (sago35) [reply examples/st7789/main.go:25]: I'm not sure if it's best practice, but I would do the following

https://github.com/tinygo-org/drivers/tree/release/examples/ili9341/pyportal_boing
https://github.com/tinygo-org/tinyfont/tree/dev/examples/initdisplay

**Comment by deadprogram** [comment]
Are there further changes needed for this PR, or is it ready to go now?

**Comment by jmstriegel** [comment]
This should be ready to go. I'll work on getting a demo based on the clue board into another pull request, based on the above feedback.

**Comment by deadprogram** [comment]
Thanks very much for this contribution @jmstriegel now merging.

**Pull request merged**